### PR TITLE
Make livereloading of watched dashboards more reliable

### PR DESCRIPTION
**Pull request by K-Phoen**
Instead of using Grafana's websocket to trigger a dashboard refresh, let's use a more crude approach and refresh the entire page.

The server-side implementation of the livereload process is taken from https://github.com/gohugoio/hugo/tree/62567d38205a61134a6822d37a534520772419f1/livereload

It has been adjusted to ensure that only dashboards that changed will be reloaded.

### CODE REVIEW

File: pkg/grizzly/server.go [hunk 1]
```diff
@@ -289,7 +289,7 @@ func (s *Server) updateWatchedResource(name string) error {
 		}
 		resources, err = s.ParseBytes(b)
 	} else {
-		resources, err = s.ParseResources(s.ResourcePath)
+		resources, err = s.ParseResources(name)
```
Comment (K-Phoen) [inline pkg/grizzly/server.go:292]: This is a drive-by fix: let's parse the resource that changed instead of re-parsing all of them.

File: internal/livereload/connection.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2015 The Hugo Authors. All rights reserved.
```
Comment (Duologic) [inline internal/livereload/connection.go:1]: Can we document the diff-patch we do here? This might make it easier to compare with upstream in the future.

**Review by Duologic — APPROVED** [review]
:+1:

File: internal/livereload/connection.go (hunk 2 above)
Comment (K-Phoen) [reply internal/livereload/connection.go:1]: I left a few comments describing where I took the code from and what I changed :)
